e2e/internal/contrasttest/contrasttest.go

```diff
@@ -81,19 +81,22 @@ func (ct *ContrastTest) Init(t *testing.T, resources []any) {
 	// Create namespace
 	namespace, err := kuberesource.ResourcesToUnstructured([]any{kuberesource.Namespace(ct.Namespace)})
 	require.NoError(err)
+	require.NoError(os.WriteFile(path.Join(ct.WorkDir, "e2e.namespace"), []byte(ct.Namespace), 0o644))
 	// Creating a namespace should not take too long.
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 	err = ct.Kubeclient.Apply(ctx, namespace...)
 	cancel()
 	require.NoError(err)
 
 	t.Cleanup(func() {
-		// Deleting the namespace may take some time due to pod cleanup, but we don't want to wait until the test times out.
-		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
-		defer cancel()
+		if os.Getenv("SKIP_UNDEPLOY") != "true" {
```

Comment:
Did you consider making this a boolean flag? That would allow for more flexibility in evaluation (e.g. `--skip-undeploy`, `--skip-undeploy=0`, `--skip-undeploy=TRUE`).